src/vm/genericdict.cpp

```diff
@@ -575,22 +631,78 @@ Dictionary::PopulateEntry(
     MethodTable *      pMT, 
     LPVOID             signature, 
     BOOL               nonExpansive, 
-    DictionaryEntry ** ppSlot)
+    DictionaryEntry ** ppSlot,
+    DWORD              dictionaryIndexAndSlot /* = -1 */)
 {
      CONTRACTL {
         THROWS;
         GC_TRIGGERS;
     } CONTRACTL_END;
 
     CORINFO_GENERIC_HANDLE result = NULL;
+    Dictionary * pDictionary = NULL;
     *ppSlot = NULL;
 
+    Module * pContainingZapModule = ExecutionManager::FindZapModule(dac_cast<TADDR>(signature));
+    Module * pReadyToRunModule = ExecutionManager::FindReadyToRunModule(dac_cast<TADDR>(signature));
```

Comment:
There tend to be expensive operations, especially for large workloads with a lot of modules (it is a liner search). Can you avoid them by passing the module around as necessary instead?